### ISSUE TITLE
Keymaster automations - not able to find Zwave integration

### ISSUE THREAD

**Issue by Scott41677** [FutureTense/keymaster#330]
Good morning all, I am having an issue with my Keymaster / Schlage Camelot Zwave lock. After reading the documents on the Keymaster wiki, I am logging a bug report as I am having an issue where the automations are throwing an error ‘A Z-Wave integration has not been configured for this Home Assistant instance’. This is with any Keymaster automation, but specifically for this issue, if I attempt to run the keymaster_synchronize_codeslot_side_garage_door_5 automation(Schlage lock is named as 'side_garage_door), that error appears in the ‘traces’ page if I manually trigger it. The automation does NOT trigger automatically when updating the code on the lovelace dashboard using the card for Code 5.

I can get a code to send to the lock using the ‘Developer tools>services>keymaster: Add code to lock’ service with no issue, but the automations do not seem to fire. I have restarted HA, reloaded the Zwave integration and reloaded Keymaster with no success. All other zwave devices and automations seem to work fine.  Has anyone else seen this issue with Keymaster?

2023-03-21 21:28:13.974 ERROR (MainThread) [custom_components.keymaster.binary_sensor] Z-Wave integration not found

**Comment by firstof9** [comment]
You have ZwaveJS working?

**Comment by Scott41677** [comment]
Yes sir. Zwave is working well and the lock has a good signal. I even
installed a zwave signal booster in close proximity to the lock in question
to help ensure a strong signal. I have ~20 zwave devices, many of which
have automations that work without issue.

Scott

**Comment by Scott41677** [comment]
![Screenshot 2023-03-22 095738](https://user-images.githubusercontent.com/45925417/226927558-8ea2b6d3-489d-4071-a10b-22cce1d04409.jpg)
![Screenshot 2023-03-22 095833](https://user-images.githubusercontent.com/45925417/226927570-efa1056f-623c-424c-9582-55908ab2ac00.jpg)

**Comment by firstof9** [comment]
That error should go away once zwavejs loads.

**Comment by Scott41677** [comment]
Hi Chris

Zwave js is loaded already and works for everything else in my mesh. In fact, I can send codes to the lock using the developer tools/services/ form but it will not work from the Lovelace form cards.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by firstof9** [comment]
Check the automation/script traces.

**Comment by Scott41677** [comment]
Hi Chris. Yes that is where I am seeing the error message in the OP.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Scott41677** [comment]
screenshot of the automation trace with the error shown. The same error is show above from the log file.  All Keymaster automations return this same error, this one is only an example to illustrate the error
![Screenshot 2023-03-22 172705](https://user-images.githubusercontent.com/45925417/227042562-269b79f3-ec2e-4d08-b298-c36ab0510c2a.png)

**Comment by Scott41677** [comment]
screenshot showing the ZwaveJS successfully running in my HA instance.
![Screenshot 2023-03-22 172900](https://user-images.githubusercontent.com/45925417/227042874-4fc133a0-56ba-48fe-9fc2-bdb86eedf706.png)

**Comment by firstof9** [comment]
Thanks for confirming, looks like there's some kind of issue checking the lock entity out in the entity registry.

@raman325 any ideas?

**Comment by raman325** [comment]
did you change the entity ID of the lock entity from what's configured in keymaster? There's no reason why it shouldn't be found as long as the IDs match

**Comment by Scott41677** [comment]
Thanks raman. I double-checked the entities and the zwave entities were listed as 'side_garage_door' as expected. However, when I looked at the entities within the Keymaster entities, they were listed as 'side_side_garage_door' for a few of them. Not all entities were listed that way, but a few were, maybe 8. 
I updated them to the 'side_garage_door' and I am in testing now. It appears that, through ZwaveJS, the user codes are updating and resetting as expected but I need to test the lock itself. I will let you know my findings as soon as I am able to do that later this afternoon.

**Comment by Scott41677** [comment]
Good afternoon, I was able to verify that after manually updating the entities and removing the extra 'side_' from the entity name, the lock page works as expected. I am now able to assign and reset codes as needed from the lovelace page. Thanks so much for that direction!

**Comment by firstof9** [comment]
Thanks for the point in the direction to fix it @raman325!

**Comment by chelming** [comment]
> did you change the entity ID of the lock entity from what's configured in keymaster? There's no reason why it shouldn't be found as long as the IDs match

I'm getting the same error in my logs (1,100 times) but everything seems to be working. I renamed some entities and did a sed replace in the packages directory for the old/new names. Is there something else I need to do? (I've done generate scripts a few times)

Same exact error though:
```
Logger: custom_components.keymaster.binary_sensor
Source: custom_components/keymaster/binary_sensor.py:49
Integration: keymaster ([documentation](https://github.com/FutureTense/keymaster), [issues](https://github.com/FutureTense/keymaster/issues))
First occurred: February 20, 2024 at 2:14:55 PM (1114 occurrences)
Last logged: 7:13:06 PM

Z-Wave integration not found
```

**Comment by raman325** [comment]
You need to update the config entry data too. You can find it in `config/.storage/core.config_entries` but be very careful, you can screw up other integrations if you update it incorrectly. If you decide to update the file by hand/via CLI, I'd back it up first just in case
